agent/pkg/proxy/iptables.go

```diff
@@ -266,3 +270,48 @@ func (proxier *Proxier) CleanRules() {
 		}
 	}
 }
+
+// CleanResidue will clean up some iptables or ip routes that may be left on the host
+func (proxier *Proxier) CleanResidue() {
+	proxier.mu.Lock()
+	defer proxier.mu.Unlock()
+
+	// clean up non-interface iptables rules
+	nonIfiRuleArgs := strings.Split(fmt.Sprintf("-p tcp -d %s -j EDGE-MESH", proxier.serviceCIDR), " ")
+	if err := proxier.iptables.DeleteRule(utiliptables.TableNAT, utiliptables.ChainPrerouting, nonIfiRuleArgs...); err != nil {
+		klog.V(4).Error(err, "Failed clean residual non-interface rule %v", nonIfiRuleArgs)
+	}
+	if err := proxier.iptables.DeleteRule(utiliptables.TableNAT, utiliptables.ChainOutput, nonIfiRuleArgs...); err != nil {
+		klog.V(4).Error(err, "Failed clean residual non-interface rule %v", nonIfiRuleArgs)
+	}
+
+	// clean up interface iptables rules
+	ifiList := []string{"docker0", "cni0"}
```

Comment:
docker0 still used?

Reply:
There may be the following residual iptables on the user's host:
```
-A PREROUTING -d 10.96.0.0/12 -i docker0 -p tcp -j EDGE-MESH
-A OUTPUT -d 10.96.0.0/12 -o docker0 -p tcp -j EDGE-MESH
```

In order to clean above residual iptables, it must be hard-coded here.

Reply:
By the way, the PR for removing docker0 dependence #36 has not been merged yet